runtime/backend/options.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+#include <executorch/runtime/core/error.h>
+#include <executorch/runtime/core/span.h>
+#include <cstddef>
+#include <cstring>
+#include <variant>
+
+namespace executorch {
+namespace runtime {
+
+// Strongly-typed option key template
+template <typename T>
+struct OptionKey {
```

Comment:
I dont think we need OptionKey as a separate class. Just use `const char *` in the `BackendOption` class.

However, the thing that worries me the most is by owning `const char*` we are giving users a footgun. They can allocate pointer to the backend option name on stack, pass the pointer around and it will go out-of-scope. If this results in segfault that might be your best option. Worst can lead to undefined behaviors.

I would suggest that we just store `const char[kMaxLegthOfBackendOptionName]`.

Reply:
You resolved this comment but I dont see it being addressed